lms/djangoapps/certificates/signals.py

```diff
@@ -1,14 +1,51 @@
 """
 Signal handler for enabling/disabling self-generated certificates based on the course-pacing.
 """
+import logging
+
 from celery.task import task
+from django.db.models.signals import post_save
 from django.dispatch import receiver
 from opaque_keys.edx.keys import CourseKey
 
-from certificates.models import CertificateGenerationCourseSetting
+from .config import waffle
+from certificates.models import CertificateGenerationCourseSetting, CertificateWhitelist
+from certificates.tasks import generate_certificate
+from courseware import courses
 from openedx.core.djangoapps.models.course_details import COURSE_PACING_CHANGE
 
 
+log = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=CertificateWhitelist, dispatch_uid="append_certificate_whitelist")
+def _listen_for_certificate_whitelist_append(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    switches = waffle.waffle()
+    # All flags enabled
+    if not switches.is_enabled(waffle.SELF_PACED_ONLY) and \
+            not switches.is_enabled(waffle.INSTRUCTOR_PACED_ONLY):
```

Comment:
minor thing: there's too much indentation on this line, I think.  Probably the two "not"s can be aligned one under the other.  Alternatively, you could use parens to split this across lines:
```python
if (
    not switches.is_enabled(waffle.SELF_PACED_ONLY) and
    not switches.is_enabled(waffle.INSTRUCTOR_PACED_ONLY)
):
    return
```

I'm not sure if that's any prettier or not.  Maybe we're alone in the universe and it doesn't matter.